index.js

```diff
@@ -1,20 +1,25 @@
-/* jshint node: true */
+/* eslint-env node */
 'use strict';
 
+var path = require('path');
+var Funnel = require('broccoli-funnel');
+var MergeTrees = require('broccoli-merge-trees');
+
 module.exports = {
   name: 'ember-countdown',
 
   included: function included(app) {
     this._super.included(app);
 
-    app.import({
-      development: app.bowerDirectory + '/countdownjs/countdown.js',
-      production: app.bowerDirectory + '/countdownjs/countdown.min.js'
-    });
-    app.import('vendor/ember-countdown.js', {
-      exports: {
-        countdown: ['default']
-      }
+    this.import('vendor/countdown.js');
+    this.import('vendor/shims/countdown.js');
+  },
+
+  treeForVendor(vendorTree) {
+    var countdownTree = new Funnel(path.dirname(require.resolve('countdown/countdown.js')), {
+      files: ['countdown.js'],
     });
+
+    return new MergeTrees([vendorTree, countdownTree]);
```

Comment:
do you mind explaining the strategy here exactly?

Reply:
So basically, the `import()` method called in the `included()` hook only works with files in the `vendor` or `bower_components` directories. In order to make it work with the `countdown.js` file from NPM we need to merge it into the vendor tree, which allows it to be imorted from `vendor/countdown.js`. 

We use `require.resolve()` method to find the path to `countdown/countdown.js` file in the `node_modules` directory since we don't know exactly where NPM will install it after it deduplicates packages. This way the dependency doesn't have to be explicitly added to the `package.json` file of the consuming application.

There's actually a nice blog post [here](https://simplabs.com/blog/2017/02/13/npm-libs-in-ember-cli.html) that explains everything in more detail.

Reply:
awesome, thanks for the explanation and link